src/framework/actions/hero.ts

```diff
@@ -0,0 +1,45 @@
+import { GameState, HeroCard, Instance, PlayerState } from "../types";
+import { isHero, queryInstances } from "../queries/common";
+import { lookupCard } from "../../data";
+import { HERO_DEATH_FATIGUE } from "../constants";
+
+export const canPlayHero = ($: GameState, hero: HeroCard): boolean => {
+  const P = $.players[$.activePlayer];
+
+  const specIndex = P.specs.indexOf(hero.spec);
+
+  if (specIndex === -1) {
+    // throw new Error("does not have chosen hero");
```

Comment:
I wish we had a better error handling system. Throwing exceptions is something I generally try to avoid. Right now, this code will return false and let hand.ts throw the exception instead, though it just throws a generic hero failure error instead of detailing the exact reason why playing the hero failed.